sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexClient.java

```diff
@@ -0,0 +1,354 @@
+     *
+     * @return the apiVersion value.
+     */
+    public SearchServiceVersion getApiVersion() {
+        return asyncClient.getApiVersion();
+    }
+
+    /**
+     * Gets the endpoint for the Azure Cognitive Search service.
+     *
+     * @return the endpoint value.
+     */
+    public String getEndpoint() {
+        return this.asyncClient.getEndpoint();
+    }
+
+    /**
+     * Gets the number of documents
+     *
+     * @return the number of documents.
+     */
+    public Long getDocumentCount() {
+        return this.getDocumentCountWithResponse(Context.NONE).getValue();
+    }
+
+    /**
+     * Gets the number of documents
+     *
+     * @param context additional context that is passed through the Http pipeline during the service call
+     * @return response containing the number of documents.
+     */
+    public Response<Long> getDocumentCountWithResponse(Context context) {
+        Mono<Response<Long>> result = asyncClient.getDocumentCountWithResponse(context);
+        return result.block();
+    }
+
+    /**
+     * Searches for documents in the Azure Cognitive Search index
+     *
+     * @param searchText search text
+     * @return A {@link PagedIterable} that iterates over {@link SearchResult} objects
+     * and provides access to the {@link SearchPagedResponse} object for each page containing HTTP response and count,
+     * facet, and coverage information.
+     */
+    public PagedIterableBase<SearchResult, SearchPagedResponse> search(String searchText) {
+        return new PagedIterableBase<>(asyncClient.search(searchText, null, null));
+    }
+
+    /**
+     * Searches for documents in the Azure Cognitive Search index
+     *
+     * @param searchText search text
+     * @param searchOptions search options
+     * @param requestOptions additional parameters for the operation.
+     * Contains the tracking ID sent with the request to help with debugging
+     * @param context additional context that is passed through the Http pipeline during the service call
+     * @return A {@link PagedIterable} that iterates over {@link SearchResult} objects
+     * and provides access to the {@link SearchPagedResponse} object for each page containing HTTP response and count,
+     * facet, and coverage information.
+     */
+    public PagedIterableBase<SearchResult, SearchPagedResponse> search(String searchText,
+                                                                       SearchOptions searchOptions,
+                                                                       RequestOptions requestOptions,
+                                                                       Context context) {
+        return new PagedIterableBase<>(asyncClient.search(searchText, searchOptions, requestOptions, context));
+    }
+
+    /**
+     * Retrieves a document from the Azure Cognitive Search index.
+     *
+     * @param key the name of the document
+     * @return document object
+     */
```

Comment:
@navalev The documentation here really needs to explain what the returned document object contains in terms of how the EDM data types map to Java types. Here is where this is explained in the .NET SDK:

https://github.com/Azure/azure-sdk-for-net/blob/8f8b4f5452c0521bfb71636730c947000e44b5f8/sdk/search/Microsoft.Azure.Search.Data/src/Customizations/Documents/IDocumentsOperations.Customization.cs#L229

These details are important; Customers need to know them.

It would also be helpful for customers if the documentation for `search`, `suggest`, and the various flavors of `index` included a remark pointing readers to the documentation here, as is done in the .NET SDK.

Reply:
updated documentation for the methods, including references to the REST APIs